src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+        /// </summary>
+        /// <param name="s">The span containing the string to parse.</param>
+        /// <param name="formats">An array of allowable formats of s.</param>
+        /// <param name="result">When this method returns, contains the DateOnly value equivalent to the date contained in s, if the conversion succeeded, or MinValue if the conversion failed. The conversion fails if the s parameter is Empty, or does not contain a valid string representation of a date. This parameter is passed uninitialized.</param>
+        /// <returns>true if the s parameter was converted successfully; otherwise, false.</returns>
+        public static bool TryParseExact(ReadOnlySpan<char> s, string [] formats, out DateOnly result) => TryParseExact(s, formats, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None, out result);
+
+        /// <summary>
+        /// Converts the specified char span of a date to its DateOnly equivalent and returns a value that indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="s">The span containing the string to parse.</param>
+        /// <param name="formats">An array of allowable formats of s.</param>
+        /// <param name="provider">An object that supplies culture-specific formatting information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that defines how to interpret the parsed date. A typical value to specify is None.</param>
+        /// <param name="result">When this method returns, contains the DateOnly value equivalent to the date contained in s, if the conversion succeeded, or MinValue if the conversion failed. The conversion fails if the s parameter is Empty, or does not contain a valid string representation of a date. This parameter is passed uninitialized.</param>
+        /// <returns>true if the s parameter was converted successfully; otherwise, false.</returns>
+        public static bool TryParseExact(ReadOnlySpan<char> s, string [] formats, IFormatProvider? provider, DateTimeStyles style, out DateOnly result)
+        {
+            result = default;
+            if ((style & ~DateTimeStyles.AllowWhiteSpaces) != 0 || formats == null)
+            {
+                return false;
+            }
+
+            DateTimeFormatInfo dtfi = DateTimeFormatInfo.GetInstance(provider);
+
+            for (int i = 0; i < formats.Length; i++)
+            {
+                string? format = formats[i];
+                if (string.IsNullOrEmpty(format))
+                {
+                    return false;
+                }
+
+                if (format.Length == 1)
+                {
+                    if (format[0] == 'o' || format[0] == 'O')
+                    {
+                        format = OFormat;
+                    }
+                    else if (format[0] == 'r' || format[0] == 'R')
+                    {
+                        format = RFormat;
+                    }
+                }
```

Comment:
I see the code many times. Make sense to convert to private static method?

Reply:
Factoring this to its own method will be a little complicated as one place is using and overwriting a string and the other place using and overwriting a span.